owltorrent-network/src/main/java/edu/rice/owltorrent/network/MessageHandler.java

```diff
@@ -0,0 +1,14 @@
+package edu.rice.owltorrent.network;
+
+import edu.rice.owltorrent.common.entity.Peer;
+
+/**
+ * Functional Interface for message handler.
+ *
+ * @author yunlyu
+ */
+public interface MessageHandler {
+  void handleMessage(PeerMessage message, PeerConnector connector) throws InterruptedException;
+
+  void removePeer(Peer peer);
```

Comment:
why does this have a removePeer method but no addPeer method? do we need the removePeer at all?

Reply:
Remove peer is a desperate attempt at adding a shutdown hook. Basically when the connection is broken, we want to close the peer connector and remove the peer from the torrent manager. But we don't want to pass the torrent manager into the peer connector